hoomd/md/pytest/test_methods.py

```diff
@@ -3,393 +3,303 @@
         # filter cannot be set after scheduling
-        npt.filter = type_A
-
-    assert npt.filter is all_
+        integrator.filter = type_A
 
-    ramp = hoomd.variant.Ramp(1, 2, 1000000, 2000000)
-    npt.kT = ramp
-    assert npt.kT is ramp
-
-    npt.tau = 10.0
-    assert npt.tau == 10.0
-
-    ramp_s = [hoomd.variant.Ramp(1.0, 4.0, 1000, 10000),
-                  hoomd.variant.Ramp(2.0, 4.0, 1000, 10000),
-                  hoomd.variant.Ramp(3.0, 4.0, 1000, 10000),
-                  hoomd.variant.Ramp(0.125, 4.0, 1000, 10000),
-                  hoomd.variant.Ramp(.25, 4.0, 1000, 10000),
-                  hoomd.variant.Ramp(.5, 4.0, 1000, 10000)]
-    npt.S = ramp_s
-    assert len(npt.S) == 6
-    for i in range(6):
-        assert npt.S[i] is ramp_s[i]
+    for parameter in method_base_params.rejected_params:
+        value = method_base_params.rejected_params[parameter]
+        with pytest.raises(AttributeError):
+            # cannot be set after scheduling
+            setattr(integrator,parameter,value)
+        assert getattr(integrator,parameter) == integrator_base_params.setup_params[parameter]
 
-    npt.tauS = 10.0
-    assert npt.tauS == 10.0
 
-    npt.box_dof = (True,False,False,False,True,False)
-    assert tuple(npt.box_dof) == (True,False,False,False,True,False)
+    for parameter in method_base_params.accepted_params:
+        value = method_base_params.accepted_params[parameter]
+        setattr(integrator,parameter,value)
+        get_value = getattr(integrator,parameter)
+        if isinstance(value,list):
+            list_length = len(get_value)
+            if parameter == 'S':
+                assert list_length == 6
+            for i in range(list_length):
+                assert get_value[i] == value[i]
+        else:
+            assert get_value == value
 
-    npt.couple = 'none'
-    assert npt.couple == 'none'
 
-    npt.rescale_all = True
-    assert npt.rescale_all
 
-    npt.gamma = 2.0
-    assert npt.gamma == 2.0
+def test_rattle_attributes(method_base_params):
 
-    assert npt.translational_thermostat_dof == (0.0, 0.0)
-    npt.translational_thermostat_dof = (0.125, 0.5)
-    assert npt.translational_thermostat_dof == (0.125, 0.5)
+    all_ = hoomd.filter.All()
+    gyroid = hoomd.md.manifold.Gyroid(N=1)
+    try:
+        integrator = method_base_params.method(**method_base_params.setup_params,filter=all_, manifold_constraint = gyroid)
+    except TypeError:
+        pass
+    else:
+        assert integrator.filter is all_
+        assert integrator.manifold_constraint == gyroid
+        assert integrator.eta == 1e-6
+
+        for parameter in method_base_params.setup_params:
+            value = method_base_params.setup_params[parameter]
+            assert getattr(integrator,parameter) == value
+
+        for parameter in method_base_params.extra_params:
+            value = method_base_params.extra_params[parameter]
+            assert getattr(integrator,parameter) == value
```

Comment:
You check this elsewhere.
```suggestion
```

Reply:
This is the first time I check this for the RATTLE integrators and I also don't check this anywhere else.

Reply:
You test this in `test_attributes`.

Reply:
But not for the rattle integrators, which work with separate C++ classes. So, for example, if you initiate `hoomd.md.methods.NVE` it calls either `TwoStepNVE` (if manifold is `None`) or `TwoStepRATTLENVE` (if manifold is `Manifold`). Therefore, this is the first time where I test the setters and getters of the rattle integrators. Admittedly, you can remove the test probably in `test_rattle_attributes`, but not in  `test_rattle_attributes_attached`.